integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/table/CarbonDropTableCommand.scala

```diff
@@ -55,6 +55,10 @@ case class CarbonDropTableCommand(
       }
       LOGGER.audit(s"Deleting table [$tableName] under database [$dbName]")
       carbonTable = CarbonEnv.getCarbonTable(databaseNameOp, tableName)(sparkSession)
+      if (carbonTable.isStreamingTable) {
+        // streaming table should acquire streaming.lock
+        carbonLocks += CarbonLockUtil.getLockObject(identifier, LockUsage.STREAMING_LOCK)
```

Comment:
better to assign to locksToBeAcquired at line 48 when initializing it.
```
   val locksToBeAcquired = if (streaming) {
       3 locks.
   } else {
       2 locks
  }
```

Reply:
Here should get LockUsage.METADATA_LOCK before invoking  CarbonEnv.getCarbonTable to get metadata.

Reply:
ok